src/opera/pge/base_pge.py

```diff
@@ -203,14 +206,40 @@ class and its inheritors, and as such, this class assumes access to the
     def __init__(self):
         self.name = None
         self.logger = None
+        self.runconfig_path = None
+        self.runconfig = RunConfig(self.runconfig_path)
+        self.catalog_metadata = {}
 
     def _run_sas_qa_executable(self):
         # TODO
         pass
 
     def _create_catalog_metadata(self):
-        # TODO
-        pass
+        """Returns the catalog metadata as a Python dictionary"""
+
+        self.cached_catalog_metadata = {
+            'PGE_Name': self.runconfig.pge_name,
+            'PGE_Version': "1.0.test",
+            'SAS_Version': "2.0.test",
+            'Input_Files': self.runconfig.input_files,
+            'Ancillary_Files': self.runconfig.get_ancillary_filenames(),
+            'Production_DateTime': get_catalog_metadata_datetime_str(self.production_datetime)
+        }
+        return self.cached_catalog_metadata
+
+    def _write_catalog_metadata(self):
+        """Create, write and validate the catalog metadata json file"""
+
+        met_dict = self._create_catalog_metadata()
+        met_filename = "test_catalog_metadata.json"
```

Comment:
Another ``#TODO`` here would be good once we've moved onto integrating file-naming conventions

Reply:
Done